prime-router/src/main/kotlin/azure/SettingsFacade.kt

```diff
@@ -60,13 +61,23 @@ class SettingsFacade(
     }
 
     override fun findReceiver(fullName: String): Receiver? {
-        val pair = Receiver.parseFullName(fullName)
-        return findSetting(pair.second, ReceiverAPI::class.java, pair.first)
+        try {
+            val pair = Receiver.parseFullName(fullName)
+            return findSetting(pair.second, ReceiverAPI::class.java, pair.first)
+        } catch (e: Exception) {
+            logger.warn("Cannot find receiver: ${e.localizedMessage} ${e.stackTraceToString()}")
+            return null
+        }
     }
 
     override fun findSender(fullName: String): Sender? {
-        val pair = Sender.parseFullName(fullName)
-        return findSetting(pair.second, SenderAPI::class.java, pair.first)
+        try {
+            val pair = Sender.parseFullName(fullName)
+            return findSetting(pair.second, SenderAPI::class.java, pair.first)
+        } catch (e: Exception) {
```

Comment:
Same catch Exception.  Try specifying the exception you are really catching if you can.